### PR TITLE
rename workspace references to symbol descriptor, clarify intro

### PR DESCRIPTION
**Pull request by ggilmore**

### CODE REVIEW

File: extension-symbol-descriptor.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
-# workspace/xreferences extension to LSP
+# symbolDescriptor extensions to LSP
```
Comment (felixfbecker) [inline extension-symbol-descriptor.md]: I think we should PascalCase SymbolDescriptor as that is the interface name

File: extension-symbol-descriptor.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
 
-The `workspace/xreferences` extension to the Language Server Protocol (LSP) enables a language server to export all of the references made from the workspace's code to its dependencies. Additionally, a new `textDocument/xdefinition` method performs the equivalent of `textDocument/definition` while returning metadata about the definition and _optionally_ a concrete location to it.
+Roughly speaking, a `symbolDescriptor` (defined fully below) encodes enough metadata about a symbol to be to uniquely identify it even when it appears in other projects (i.e. as a dependency).
```
Comment (felixfbecker) [inline extension-symbol-descriptor.md]: I would change this to be more accurate, omitting "roughly speaking":

_A SymbolDescriptor contains metadata about a symbol to identify it as uniquely as possible._

File: extension-symbol-descriptor.md [hunk 3]
```diff
@@ -1,6 +1,11 @@
...
+
+The `symbolDescriptor` extensions to the Language Server Protocol (LSP) allow a language server to able to interact with `symbolDescriptors` via the following methods:
+- The `textDocument/xdefinition` method is similar to `textDocument/definition` - it returns a `symbolDescriptor` and _optionally_ a concrete location for the definition
```
Comment (felixfbecker) [inline extension-symbol-descriptor.md]: The most important part here is that `xdefinition` is a method to get a SymbolDescriptor for a symbol at a location, the fact that the definition is included is only a performance optimization not worth mentioning in a summary.

File: extension-symbol-descriptor.md [hunk 4]
```diff
@@ -1,6 +1,11 @@
...
+- The `workspace/xreferences` method locates project wide references to a symbol, given a `symbolDescriptor`
+- The `workspace/symbol` method allows you to search for an instance of a symbol, given a `symbolDescriptor`
```
Comment (felixfbecker) [inline extension-symbol-descriptor.md]: I find "instance" to be confusing here, this searches for definitions of a symbol. Note that this is not the sole purpose of workspace/symbol, the method is part of the official spec. This extension just adds a filter parameter to it.

**Review by keegancsmith — COMMENTED** [review]
From reading this again, it isn't clear to me how `hints` are used. It seems that is input to the LS, but where do the hints come from? I assume something the LS returns contains hints?

Otherwise LGTM + agree with Felix's comments.

**Comment by ggilmore** [comment]
@keegancsmith I'm actually not sure about the semantics of `hints` off the top of my head either

@felixfbecker PTAL

**Review by felixfbecker — APPROVED** [review]

**Comment by felixfbecker** [comment]
Yeah I forgot what hints was about. Something with Go knowing which file paths to search.

**Pull request merged**